datafusion/proto/src/bytes/mod.rs

```diff
@@ -87,8 +88,8 @@ pub trait Serializeable: Sized {
 impl Serializeable for Expr {
     fn to_bytes(&self) -> Result<Bytes> {
         let mut buffer = BytesMut::new();
-        let protobuf: protobuf::LogicalExprNode = self
-            .try_into()
+        let extension_codec = DefaultLogicalExtensionCodec {};
+        let protobuf: protobuf::LogicalExprNode = serialize_expr(self, &extension_codec)
```

Comment:
Would this be OK?
I think `DefaultLogicalExtensionCodec` may need to implement some functions it didn't implement yet. I will also look into it.

Reply:
Yes I think this is fine. These extension methods can be used for vanilla serialization and then if user needs custom stuff with a custom codec they can use the lower-level apis